build/utils/process_runner.py

```diff
@@ -23,11 +23,17 @@ def run_process_and_log(list_of_args):
                                stderr=subprocess.PIPE)
     process_output, process_error = process.communicate()
     exit_status = process.returncode
-    if process_output:
+    # Typically, an exit status of 0 indicates that the process ran
+    # successfully. However, this is for example found not to be true when
+    # not have the right permission with activemq
+    if exit_status != 0 or 'Permission denied' in process_error.decode("utf-8"):
+        # log both process_output and process_error, since process_output
+        # may also info that help with diagnose the problem
         BUILD_LOGGER.logger.info(process_output)
-    if exit_status != 0:
         BUILD_LOGGER.logger.error(process_error)
         return False
+    if process_output:
+        BUILD_LOGGER.logger.info(process_output)
```

Comment:
Could we move this into a seperate method so we can make the methods more concise and easier to debug if problems ever arise given this setup is untested. 
e.g.

```python
def log_process(process_output, process_error, exit_status):
    """ 
    Typically, an exit status of 0 indicates that the process ran
    successfully. However, this is for example found not to be true when
    not have the right permission with activemq
    :param process_output: (sub-process) <Param Description>
    :param process_error: (sub-process) <Param Description>
    :param exit_status: (sub-process) <Param Description>
    :return: (bool) True or None
    """

    if exit_status != 0 or 'Permission denied' in process_error.decode("utf-8"):
        # log both process_output and process_error, since process_output
        # may also info that help with diagnose the problem
        BUILD_LOGGER.logger.info(process_output)
        BUILD_LOGGER.logger.error(process_error)
        return False
    if process_output:
        BUILD_LOGGER.logger.info(process_output)
    return True


def run_process_and_log(list_of_args):
    """
    Call a process using Popen and logs output to file
    :param list_of_args: list of arguments for Popen
    :return: (bool) True or None
    """
    process = subprocess.Popen(list_of_args,
                               stdout=subprocess.PIPE,
                               stderr=subprocess.PIPE)
    process_output, process_error = process.communicate()
    exit_status = process.returncode

    return log_process(process_output, process_error, exit_status)


def run_process_with_shell(list_of_args):
    """
    Run a process using the shell true command and do not log the output
    :param list_of_args: list of args for Popen
    :return:
    """
    subprocess.call(list_of_args, shell=True)
```

Reply:
Thanks @JackEAllen for suggestion, I have added time hint to return value, but because let as one method for now and your suggestion for splitting can be added when we get testing of deployment setup